cmd/node/node.go

```diff
@@ -1164,3 +1170,35 @@ func decodeLoggers(cfg config.LoggerConfig) (map[string]string, error) {
 	}
 	return rst, nil
 }
+
+func (app *App) checkAndStoreGenesisConfig() error {
+	genesisConfigPath := filepath.Join(app.Config.Genesis.Path, config.DefaultGenesisConfigFileName)
+	if exists := filesystem.PathExists(genesisConfigPath); exists {
+		genesisData, err := ioutil.ReadFile(genesisConfigPath)
+		if err != nil {
+			return fmt.Errorf("failed to read stored genesis config file: %s", err)
+		}
+
+		var payload config.GenesisConfig
+
+		if err = json.Unmarshal(genesisData, &payload); err != nil {
+			return fmt.Errorf("failed to unmarshal stored genesis config file: %w", err)
+		}
+		if err = app.Config.Genesis.Compare(&payload); err != nil {
+			return fmt.Errorf("failed to compare stored genesis config file: %w", err)
```

Comment:
the txt is misleading. how about "genesis config changed from previous run"?

Reply:
yes message is bad. it also needs to specify location of persisted config file

Reply:
fixed